pkg/cmd/roachtest/test_runner.go

```diff
@@ -2116,3 +2134,47 @@ func getTestParameters(t *testImpl, c *clusterImpl, createOpts *vm.CreateOpts) m
 
 	return clusterParams
 }
+
+// PollPreemptionInterval is how often to poll for preempted VMs.
+var PollPreemptionInterval = 5 * time.Minute
+
+var preemptionMonitor = monitorForPreemptedVMs
+
+func monitorForPreemptedVMs(
+	ctx context.Context,
+	t test.Test,
+	c cluster.Cluster,
+	l *logger.Logger,
+	preemptedVMCh chan vm.PreemptedVM,
+) {
+	if c.IsLocal() || !c.Spec().UseSpotVMs {
+		close(preemptedVMCh)
+		return
+	}
+
+	go func() {
+		pollCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
```

Comment:
Looks rather circuitous. We can only get out of the loop if the parent context (i.e., `ctx`) is cancelled; so this `cancel` is technically not needed or used.